app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,68 @@
+class VotesController < ApplicationController
+  before_action :review, only: [:create, :update]
+  before_action :park, only: [:create, :update]
+
+  def create
+    if params[:vote] == "up"
+      vote = @review.votes.new(vote: 1)
+      @review.increment(:vote_count)
+    elsif params[:vote] == "down"
+      vote = @review.votes.new(vote: -1)
+      @review.increment(:vote_count, -1)
+    else
+      vote = @review.votes.new
+    end
+
+    vote.user = current_user
+    @review.save
+
+    if vote.save
+      flash[:success] = "You have #{params[:vote]} voted successfully!"
+      redirect_to park_path(@park)
+    else
+      flash[:notice] = vote.errors.full_messages
+      render 'parks/show'
+    end
+  end
+
+  def update
+    vote = Vote.where(review: @review, user: current_user).first
+
+    if vote.vote == 0 && params[:vote] == "up"
+      vote.update(vote: 1)
+      @review.increment(:vote_count).save
```

Comment:
You want to make a decision based on the return value of both updating the review and vote. Thus change this to the following.
```
value = 1
@review.increment(:vote_count)
```
Do the same for the other clauses below